base/channels.jl

```diff
@@ -553,6 +664,127 @@ function take_unbuffered(c::Channel{T}) where T
     end
 end
 
+"""
+    take!(c::Channel, n::Integer[, buffer::AbstractVector])
+
+Take at most `n` items from the [`Channel`](@ref) `c` and return them in a `Vector`.
+If `buffer` is provided, it will be used to store the result. If the channel closes before
+`n` items are taken, the result will contain only the items that were available. For
+buffered channels, this operation requires fewer `lock` operations than individual `take!(c)`s.
+
+# Examples
+
+```jldoctest
+julia> c = Channel(3);
+
+julia> append!(c, 1:3);
+
+julia> take!(c, 3, zeros(Int, 3))
+[1,2,3]
+```
+
+!!! compat "Julia 1.12"
+    Requires at least Julia 1.12.
+"""
+function take!(c::Channel{T}, n::Integer) where {T}
+    return _take(c, n, Vector{T}(undef, n))
+end
+function take!(c::Channel{T}, n::Integer, buffer::AbstractVector{T2}) where {T2,T<:T2}
```

Comment:
I am limiting the buffer to `AbstractVector` because, if the channel closes, we need to return fewer than `n` elements. In that case, we need to resize the buffer. So that precludes `Memory` or `Matrix` as buffer types. I'm not sure if there is a more general way to accept more buffer types here.

Reply:
`AbstractVector` is fine, but do note that that `Memory` is also an abstractvector.

Reply:
Should I do a check to make sure the buffer is `resize!`able, then?